### PR TITLE
Build and deploy

### PR DESCRIPTION
**Pull request by iocanel**
Currently, to build an image using docker you need to use`-Dap4k.build=true` for openshift you need `-Dap4k.deploy=true`.

This pull request aligns handling of `ap4k.deploy` and `ap4k.build` accross modules, so that build triggers a build and deploy triggers a deploy.

Note: for openshift build, will also apply:
- BuildConfig
- ImageStream
- Secret (not sure if we really need that)

Also fixes some minor issues found along the way:
- broken test
- using a better default for group name.

### CODE REVIEW

File: annotations/openshift-annotations/src/main/java/io/ap4k/openshift/configurator/ApplySourceToImageHook.java [hunk 1]
```diff
@@ -23,10 +23,13 @@
...
 
+  private static final String AP4K_BUILD = "ap4k.build";
   private static final String AP4K_DEPLOY = "ap4k.deploy";
 
   @Override
   public void visit(S2iConfigFluent config) {
-    config.withAutoDeployEnabled(Boolean.parseBoolean(System.getProperty(AP4K_DEPLOY, String.valueOf(config.isAutoDeployEnabled()))));
+    config
+      .withAutoBuildEnabled(Boolean.parseBoolean(System.getProperty(AP4K_BUILD, String.valueOf(config.isAutoBuildEnabled()))))
+      .withAutoBuildEnabled(Boolean.parseBoolean(System.getProperty(AP4K_DEPLOY, String.valueOf(config.isAutoBuildEnabled()))));
```
Comment (geoand) [inline annotations/openshift-annotations/src/main/java/io/ap4k/openshift/configurator/ApplySourceToImageHook.java]: Isn't this wrong?

File: annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java [hunk 2]
```diff
@@ -17,29 +17,53 @@
...
 import java.io.File;
-import java.io.FileNotFoundException;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
 
 public class OcBuildHook extends ProjectHook {
```
Comment (geoand) [inline annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java]: Wouldn't it be better to name this `JavaBuildHook`?

File: annotations/openshift-annotations/src/main/java/io/ap4k/openshift/configurator/ApplySourceToImageHook.java (hunk 1 above)
Comment (geoand) [reply annotations/openshift-annotations/src/main/java/io/ap4k/openshift/configurator/ApplySourceToImageHook.java]: I mean shouldn't be `withAutoDeployEnabled`?

File: annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java (hunk 2 above)
Comment (iocanel) [reply annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java]: It uses oc to perform the actual build.

File: annotations/openshift-annotations/src/main/java/io/ap4k/openshift/configurator/ApplySourceToImageHook.java (hunk 1 above)
Comment (iocanel) [reply annotations/openshift-annotations/src/main/java/io/ap4k/openshift/configurator/ApplySourceToImageHook.java]: correct. let me fix that.

File: annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java (hunk 2 above)
Comment (geoand) [reply annotations/openshift-annotations/src/main/java/io/ap4k/openshift/hook/OcBuildHook.java]: Yeah... stupid me :P. Looking at the `init` method threw me off, I didn't see the `run` method

**Pull request merged**